modules/core/src/main/scala/internal/decoding.scala

```diff
@@ -132,7 +154,7 @@ object decoding {
       .foldMap { alt => (v: DynamoValue) =>
         alt.caseSchema.read(v).map(alt.prism.inject).toOption
       }
-      .andThen(_.toRight(ReadError()))
+      .andThen(_.toRight(ReadError("")))
```

Comment:
Was unclear what the error message should be here

Reply:
It's something like "data doesn't match any of the alternatives", like some sort of pattern matching error